actionpack/lib/action_dispatch/routing/inspector.rb

```diff
@@ -206,6 +206,78 @@ def widths(routes)
           end
       end
 
+      class Colorized < Base
+        CLEAR   = "\e[0m"
+
+        # Colors
+        BLACK   = "\e[1;90m"
+        BLUE    = "\e[1;94m"
+        RED     = "\e[1;91m"
+        YELLOW  = "\e[1;93m"
+        WHITE   = "\e[1;97m"
+
+        def initialize(width: IO.console_size[1])
+          @width = width
+          super()
+        end
+
+        def section_title(title)
+          @buffer << "\n#{title}:"
+        end
+
+        def section(routes)
+          @buffer << draw_section(routes)
+        end
+
+        def header(routes)
+          @buffer << draw_header(routes)
+        end
+
+        private
+          def draw_section(routes)
+            routes.map do |route|
+              verb = "     #{route[:verb]}"
+              path = "#{route[:path]} ".rjust(10 - route[:verb].size + route[:path].size)
+              name_and_reqs = " #{name_and_reqs(route)}     ".rjust(@width - verb.size - path.size, ".")
+
+              verb = colorize_verb(verb)
+              path = colorize_path(path)
+              name_and_reqs = "#{BLACK}#{name_and_reqs}#{CLEAR}"
+
+              "#{verb}#{path}#{name_and_reqs}"
+            end
+          end
+
+          def colorize_verb(verb)
+            case verb
+            when %r{GET}
+              "#{BLUE}#{verb}#{CLEAR}"
+            when %r{HEAD}
+              "#{BLACK}#{verb}#{CLEAR}"
+            when %r{PUT|POST|PATCH}
+              "#{YELLOW}#{verb}#{CLEAR}"
+            when %r{DELETE}
+              "#{RED}#{verb}#{CLEAR}"
+            else
+              verb
+            end
+          end
+
+          def colorize_path(path)
+            path.gsub(/\/(:[a-z]*)/, "/#{YELLOW}\\1#{CLEAR}")
```

Comment:
This doesn't fully cover a param. For example, your screenshot only highlights `:inbound` of `:inbound_email_id`. Parameters can also include numbers, underscores, capitals, and possibly other characters.

Reply:
Great catch, thanks! I was trying some ways to solve this, what do you think about this?

```ruby
path.gsub(/\/(:[\w\d]*)/, "/#{YELLOW}\\1#{CLEAR}")
```

Here before and after that change:

<img width="905" alt="image" src="https://user-images.githubusercontent.com/16840674/171550602-6c72f888-741e-4db4-b529-219714404866.png">